### ISSUE TITLE
Rubocop autocorrecting "#$1" produces incorrect code 

### ISSUE THREAD

**Issue by ghost** [rubocop/rubocop#1465]
Big fan of Rubocop, we discovered a bug in it's autocorrect feature with certain code:

``` Bash
$ rubocop -V
warning: parser/current is loading parser/ruby21, which recognizes
warning: 2.1.5-compliant syntax, but you are running 2.1.4.
0.27.1 (using Parser 2.2.0.pre.8, running on ruby 2.1.4 x86_64-darwin14.0)
$ rubocop -a interpolation.rb
```

``` Ruby
# interpolation.rb
# Before code:
'foo' =~ /fo(o)/
puts "Match: #$1"

# After code:
'foo' =~ /fo(o)/
puts 'Match: #Regexp.last_match[1]'
```

It should have been produced:

``` Ruby
'foo' =~ /fo(o)/
puts "Match: #{Regexp.last_match[1]}"
```
